scene/resources/packed_scene.cpp

```diff
@@ -161,6 +172,7 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 	bool gen_node_path_cache = p_edit_state != GEN_EDIT_STATE_DISABLED && node_path_cache.is_empty();
 
 	HashMap<Ref<Resource>, Ref<Resource>> resources_local_to_scene;
+	HashMap<Node *, HashMap<Ref<Resource>, Ref<Resource>>> resources_local_to_sub_scene; // Record the mappings in sub-scenes.
```

Comment:
Do we want to create `typedef`s for these types, they're very cumbersome

Reply:
We rarely use typedefs.